tornado/test/ioloop_test.py

```diff
@@ -694,6 +694,60 @@ def test_timedelta(self):
         self.assertEqual(pc.callback_time, expected_callback_time)
 
 
+class TestPeriodicCallbackAsync(AsyncTestCase):
```

Comment:
I think we want tests that cover the overrun behavior (since this is the part that's changing for decorated coroutines between this change and the prior implementation)